include/mxnet/ndarray.h

```diff
@@ -202,15 +192,23 @@ class NDArray {
    * nnz that is smaller than nnz1+nnz2. Therefore, the storage shape's size
    * needs to be shrunk from nnz1+nnz2 to nnz.
    */
-  inline void SetStorageShape(const TShape& sshape) {
+  inline void set_storage_shape(const TShape& sshape) {
     CHECK(storage_type() != kDefaultStorage);
     ptr_->storage_shape = sshape;
   }
 
   /*!
    * \return the shape of aux data at ith index. If it doesn't exist, return an empty one.
    */
-  inline const TShape aux_shape(size_t i) const {
+  inline size_t aux_shape_count() const {
```

Comment:
It's determined by storage_type only. In C++, if an attribute is not determined by a maintained state of a class, it's preferred to use a free function.